fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -307,6 +307,22 @@ private Type getCmpType() {
             return Type.LARGEINT;
         }
 
+        // Try parse varchar to bigint and decide type
```

Comment:
would better add a comment to describe why do we want to convert string to bigint. add note this is compatible with Mysql